File: DAL/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data;

namespace dal
{
    public interface IRepository : IDisposable
    {
        string Nombre {get; set;}
        int CommandTimeout { get; set; }
        void BeginTransaction();

        void AddParameter(string name, object value);

        int ExecuteNonQuery();
        object ExecuteReturnValue();
        object ExecuteScalar();
        DataTable ExecuteDataTable();
        DataSet ExecuteDataSet();

        void Commit();
        void Rollback();

        String printSP();
        Boolean IsDisposed { get; }

    }
}

File: TPVentaZapatillas/DAL/IRepository.vb
'TODO: Patron Proxy -> Subject
Public Interface IRepository
    Inherits IDisposable

    Sub Conectar()
    Sub Desconectar()
    Sub BeginTransaction()
    Sub BackUp(path As String)
    Sub Restore(path As String)
    Property Nombre() As String
    Property Timeout() As Integer
    ReadOnly Property IsDisposed() As Boolean
    Sub AddParameter(name As String, value As Object)
    Sub AddClient()
    Function ExecuteNonQuery() As Integer
    Function ExecuteScalar() As Object
    Function ExecuteDataTable() As DataTable
    Function ExecuteDataSet() As DataSet
    Function ExecuteReturnValue() As Object
    Sub Commit()
    Sub Rollback()

End Interface
